vllm/transformers_utils/config.py

```diff
@@ -100,15 +100,26 @@ def file_or_path_exists(model: Union[str, Path], config_name: str,
 
     # NB: file_exists will only check for the existence of the config file on
     # hf_hub. This will fail in offline mode.
-    try:
-        return file_exists(model,
-                           config_name,
-                           revision=revision,
-                           token=HF_TOKEN)
-    except huggingface_hub.errors.OfflineModeIsEnabled:
-        # Don't raise in offline mode, all we know is that we don't have this
-        # file cached.
-        return False
+
+    # Call HF to check if the file exists, with 3 retries and exponential backoff
+    max_retries = 3
+    retry_delay = 2
+    for attempt in range(max_retries):
+        try:
+            return file_exists(model,
+                             config_name,
+                             revision=revision,
+                             token=HF_TOKEN)
+        except huggingface_hub.errors.OfflineModeIsEnabled:
+            # Don't raise in offline mode, all we know is that we don't have this
+            # file cached.
+            return False
+        except Exception as e:
+            logger.error(f"Error checking file existence: {e}")
+            if attempt == max_retries - 1:
+                raise
+            time.sleep(retry_delay)
+            retry_delay *= 2
```

Comment:
Hmm - doesn't this mean we're printing the `logger.error` for the first two retries but not actually raising? This can be misleading to users sometimes.

Can we do something like the below?
```suggestion
        except Exception as e:
            if attempt == max_retries - 1:
             logger.error(f"Maximum amount of retries {(max_retries)} reached; Error checking file existence: {e}")
                raise
            logger.warning(f"Error checking file existence, retrying {(attempt + 1)}: {e}")
            time.sleep(retry_delay)
            retry_delay *= 2
```

Reply:
ah yes this makes sense.. fixing